src/test/java/cloud/fogbow/ras/core/processors/UnableToCheckStatusProcessorTest.java

```diff
@@ -1,313 +1,308 @@
+    }
+
+    // test case: When invoking the processUnableToCheckStatusOrder method with an
+    // instance that failed after a successful request, it must change the order
+    // state to FAILED_AFTER_SUCCESSFUL_REQUEST.
+    @Test
+    public void testProcessUnableToCheckStatusOrderChangeStateToFailed() throws FogbowException {
+        // set up
+        Order order = createLocalOrder(getLocalMemberId());
+        order.setInstanceId(BaseUnitTests.FAKE_INSTANCE_ID);
+        order.setOrderStateInTestMode(OrderState.UNABLE_TO_CHECK_STATUS);
+        this.unableToCheckStatus.addItem(order);
+        
+        OrderInstance orderInstance = new ComputeInstance(BaseUnitTests.FAKE_INSTANCE_ID);
+        orderInstance.setHasFailed();
+
+        Mockito.doReturn(orderInstance).when(this.cloudConnector).getInstance(Mockito.any(Order.class));
+
+        // exercise
+        this.processor.processUnableToCheckStatusOrder(order);
+
+        // verify
+        Assert.assertEquals(OrderState.FAILED_AFTER_SUCCESSFUL_REQUEST, order.getOrderState());
+    }
+
+    // test case: When invoking the processUnableToCheckStatusOrder method and an
+    // error occurs while trying to get an instance from the cloud provider, an
+    // UnavailableProviderException will be throw.
+    @Test(expected = UnavailableProviderException.class) // Verify
+    public void testProcessUnableToCheckStatusOrderThrowsUnavailableProviderException() throws FogbowException {
+        // set up
+        Order order = createLocalOrder(getLocalMemberId());
+        order.setInstanceId(BaseUnitTests.FAKE_INSTANCE_ID);
+        order.setOrderStateInTestMode(OrderState.UNABLE_TO_CHECK_STATUS);
+        this.unableToCheckStatus.addItem(order);
+
+        Mockito.doThrow(new UnavailableProviderException()).when(this.cloudConnector)
+                .getInstance(Mockito.any(Order.class));
+
+        // exercise
+        this.processor.processUnableToCheckStatusOrder(order);
+    }
+
+    // test case: When calling the processUnableToCheckStatusOrder method and the
+    // order instance is not found, it must change the order state to
+    // FAILED_AFTER_SUCCESSFUL_REQUEST.
+    @Test
+    public void testProcessUnableToCheckStatusOrderWithInstanceNotFound() throws FogbowException {
+        // set up
+        Order order = createLocalOrder(getLocalMemberId());
+        order.setInstanceId(BaseUnitTests.FAKE_INSTANCE_ID);
+        order.setOrderStateInTestMode(OrderState.UNABLE_TO_CHECK_STATUS);
+        this.unableToCheckStatus.addItem(order);
+
+        Mockito.doThrow(new InstanceNotFoundException()).when(this.cloudConnector)
+                .getInstance(Mockito.any(Order.class));
+
+        // exercise
+        this.processor.processUnableToCheckStatusOrder(order);
+
+        // verify
+        Assert.assertEquals(OrderState.FAILED_AFTER_SUCCESSFUL_REQUEST, order.getOrderState());
+    }
+
+    // test case: When calling the processUnableToCheckStatusOrder method with a
```

Comment:
why shouldn't it be changed?

Reply:
Because remoted orders are monitored by the remote provider and change state when that provider notifies state changes.

Reply:
@raonismaneoto can this conversation be resolved or still open to you?